ni_measurementlink_service/_internal/grpc_servicer.py

```diff
@@ -83,7 +83,35 @@ def time_remaining(self) -> float:
 
     def abort(self, code: grpc.StatusCode, details: str) -> None:
         """Aborts the RPC."""
-        self._grpc_context.abort(code, details)
+        try:
+            self._grpc_context.abort(code, details)
+        except Exception as e:
+            # If gRPC raises an empty exception, replace it with an RpcError.
+            # This allows our logging interceptors to query the code/details.
+            if type(e) is Exception:
+                raise CustomRpcError(code, details) from e
+            raise e
```

Comment:
in Python -> if you manually raise an exception, it rasies with this starting as the stack trace. If you put an empty `raise` that will "re-raise" without altering the stack trace

```suggestion
            raise
```

Reply:
Thank you, I included this in my suggestions but didn't take the time to explain it.